docs/user/unsupported/mbstyle.rst

```diff
@@ -19,14 +24,544 @@ Quick example of module usage (API change):
+
+    A *value* (and *v0, ..., vn* for set operators) must be a string, number, or boolean to compare the property value against.
+
+    Set membership filters are a compact and efficient way to test whether a field matches any of multiple values.
+
+    The comparison and set membership filters implement strictly-typed comparisons; for example, all of the following evaluate to false: 0 < "1", 2 == "2", "true" in [true, false].
+
+    The "all", "any", and "none" filter operators are used to create compound filters. The values *f0, ..., fn* must be filter expressions themselves.
+
+    .. code-block:: json
+
+      ["==", "$type", "LineString"]
+
+    This filter requires that the class property of each feature is equal to either "street_major", "street_minor", or "street_limited".
+
+    .. code-block:: json
+
+      ["in", "class", "street_major", "street_minor", "street_limited"]
+
+    The combining filter "all" takes the three other filters that follow it and requires all of them to be true for a feature to be included: a feature must have a class equal to "street_limited", its admin_level must be greater than or equal to 3, and its type cannot be Polygon. You could change the combining filter to "any" to allow features matching any of those criteria to be included - features that are Polygons, but have a different class value, and so on.
+
+    .. code-block:: json
+
+      [
+        "all",
+        ["==", "class", "street_limited"],
+        [">=", "admin_level", 3],
+        ["!in", "$type", "Polygon"]
+      ]
+
+MapBox Style Grammar
+^^^^^^^^^^^^^^^^^^^^
+  *JSON does not allow for comments within the data therefore comments will be noted through the placement of the comment between open < and close > angle brackets. All properties are optional unless otherwise noted as Requried*
+
+  Root Properties
+  '''''''''''''''
+
+  .. code-block:: json
+
+    {
+      "version": 8, <Required>
+      "name": "Mapbox Streets",
+      "sprite": "mapbox://sprites/mapbox/streets-v8",
+      "glyphs": "mapbox://fonts/mapbox/{fontstack}/{range}.pbf",
+      "sources": {...}, <Required>
+      "layers": [...] <Required>
+    }
+
+  layers
+  ''''''
+  Layers are drawn in the order they appear in the layer array. Layers have two additional properties that determine how data is rendered: *layout* and *paint*
+
+  Background Layer definition
+  '''''''''''''''''''''''''''
+  .. code-block:: json
+
+    {
+      "layers" : [
+        {
+          "id": "backgroundcolor",
+          "type": "background",
+          "source": "test-source",
+          "source-layer": "test-source-layer",
+          "layout": {
+            "visibility": "visible"
+          },
+          "paint": {
+            "background-opacity": 0.45,
+            "background-color": "#00FF00"
+          }
+        }
+      ]
+    }
+
+  *background-color* is disabled by the presence of *background-pattern*
```

Comment:
We need a compatibility table here that says we do not support background layer yet.